crates/re_ui/src/list_item2/property_content.rs

```diff
@@ -176,35 +175,67 @@ impl ListItemContent for PropertyContent<'_> {
+        // │ ◀───────────state.left_column_width────────────▶│┌──COLUMN_SPACING           │
+        // │                                                  ▼                           │
+        // │                       ◀──────────────CONTENT────┼──────────────────────────▶ │
+        // │ ┌ ─ ─ ─ ─ ┬ ─ ─ ─ ─ ┬ ┬────────┬─┬─────────────┬─┬─────────────┬─┬─────────┐ │
+        // │                       │        │ │             │││             │ │         │ │
+        // │ │         │         │ │        │ │             │ │             │ │         │ │
+        // │   INDENT       ▼      │  ICON  │ │    LABEL    │││    VALUE    │ │   BTN   │ │
+        // │ │         │         │ │        │ │             │ │             │ │         │ │
+        // │                       │        │ │             │││             │ │         │ │
+        // │ └ ─ ─ ─ ─ ┴ ─ ─ ─ ─ ┴ ┴────────┴─┴─────────────┴─┴─────────────┴─┴─────────┘ │
+        // │ ▲                     ▲         ▲               │               ▲            │
+        // │ └──state.left_x       │         └───────────────────────────────┤            │
+        // │                       │                         ▲               │            │
+        // │       content_left_x──┘           mid_point_x───┘     text_to_icon_padding   │
+        // │                                                                              │
+        //
+        // content_indent = content_left_x - state.left_x
+        // left_column_width = content_indent + icon_extra + label_width + COLUMN_SPACING/2
+
+        let state = super::StateStack::top(ui.ctx());
+
+        let content_left_x = context.rect.left();
+        // Total indent left of the content rect. This is part of the left column width.
+        let content_indent = content_left_x - state.left_x;
+        let mid_point_x = state.left_x
+            + state
+                .left_column_width
+                .unwrap_or_else(|| content_indent + (context.rect.width() / 2.).at_least(0.0));
+
+        let icon_extra = if icon_fn.is_some() {
+            ReUi::small_icon_size().x + ReUi::text_to_icon_padding()
+        } else {
+            0.0
+        };
 
         let icon_rect = egui::Rect::from_center_size(
             context.rect.left_center() + egui::vec2(ReUi::small_icon_size().x / 2., 0.0),
             ReUi::small_icon_size(),
         );
 
-        let mut label_rect = egui::Rect::from_min_size(
-            context.rect.left_top(),
-            egui::vec2(column_width, context.rect.height()),
+        // TODO(#6179): don't reserve space for action button if none are ever used in the current
+        // scope.
+        let action_button_rect = egui::Rect::from_center_size(
+            context.rect.right_center() - egui::vec2(ReUi::small_icon_size().x / 2., 0.0),
+            ReUi::small_icon_size() + egui::vec2(1.0, 1.0), // padding is needed for the buttons
```

Comment:
why is it needed? And can we not hard-code it please?

Reply:
Moved the rect down to where its needed and cleaned up the code.